File: web_api/schemas/rag.py
from pydantic import BaseModel
from typing import Any


class RagJob(BaseModel):
    job_id: str
    shop_id: str
    source_type: str
    domain: str
    version: str
    status: str
    chunk_count: int
    embedded_count: int
    indexed_count: int
    error_summary: str | None = None
    created_at: str
    finished_at: str | None = None
    dry_run: bool = True
    calls_ollama: bool = False
    connects_pgvector: bool = False


class RagJobCreate(BaseModel):
    shop_id: str
    source_type: str
    domain: str
    version: str = "draft"
    dry_run: bool = True


class RagDebugRequest(BaseModel):
    shop_id: str
    query: str
    domain: str = "product_catalog"
    version: str = "real-product-v1"
    top_k: int = 5
    goods_id: str | None = None


class RagDebugHit(BaseModel):
    chunk_id: str
    content: str
    score: float
    domain: str
    source_type: str
    source_id: str
    version: str
    metadata: dict[str, Any]


class RagDebugResponse(BaseModel):
    query: str
    shop_id: str
    domain: str
    version: str
    hits: list[RagDebugHit]
    connects_pgvector: bool = False
    calls_ollama: bool = False
    warning: str | None = None

File: web_frontend/src/api/rag.ts
import { apiGet, apiPost } from "./client";

export type RagJob = {
  job_id: string;
  shop_id: string;
  source_type: string;
  domain: string;
  version: string;
  status: string;
  chunk_count: number;
  embedded_count: number;
  indexed_count: number;
  dry_run: boolean;
};

export function getRagJobs() {
  return apiGet<{ items: RagJob[]; total: number }>("/api/rag/jobs");
}

export function createRagJob(body: Partial<RagJob>) {
  return apiPost<RagJob>("/api/rag/jobs", body);
}

export type RagDebugRequest = {
  shop_id: string;
  query: string;
  domain: string;
  version?: string;
  top_k: number;
  goods_id?: string;
};

export type RagDebugHit = {
  chunk_id: string;
  content: string;
  score: number;
  domain: string;
  source_type: string;
  source_id: string;
  version: string;
  metadata: Record<string, unknown>;
};

export type RagDebugResponse = {
  query: string;
  shop_id: string;
  domain: string;
  version: string;
  hits: RagDebugHit[];
  connects_pgvector: boolean;
  calls_ollama: boolean;
  warning?: string | null;
};

export function retrieveDebug(body: RagDebugRequest) {
  return apiPost<RagDebugResponse>("/api/rag/retrieve-debug", body);
}
